### PR TITLE
Accept User Input

### LINKED ISSUE #29: [F] Accept User Input
As a user, I want to be able to input a list of columns and their values to be checked against the database.

AC:
- Have four input fields
  - Table to look in
  - Column to look at (can be search dropdown or input, implementation detail)
https://www.w3schools.com/howto/howto_js_filter_dropdown.asp
  - Condition to match by (contains, exactly, etc.)
  - Value to compare to
- Replaces "view code" area on Execute screen
For each rule in rule table,
- User can add additional column fields using a '+' button
- Each column field has a corresponding field to enter values

### PR DESCRIPTION
**Pull request by alex-hobby**
# Description

PR for adding test conditions.

Workflow:
- select table: auto-populated based on current db
- select test type: table, string, boolean, id, number
- populate inputs via dropdowns - available options are restricted in some cases, such as table exists, table count, and boolean comparisons
- can add / remove from list, which is stored in an object in Execute.tsx

Execution:
- when pressed, executes the test
- output to later be passed to results screen once merged

## Link to the issue you are closing

Fixes #29 

## Screenshot or other testing that you have completed

https://user-images.githubusercontent.com/29695042/197437246-14ccee94-2fdb-4947-88b1-7918bfc42c0b.mov

# Checklist:

- [ ] ESLint passes
- [ ] Prettier passes
- [ ] Included comments where necessary
- [ ] Updated README if needed
- [ ] Verified that code runs and generates no errors/warnings

### CODE REVIEW

File: src/renderer/components/execute/conditions/ConditionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        break;
+    }
+  };
+
+  useEffect(() => {
+    const getColumns = async () => {
+      const cols = await window.procedures.ipcRenderer.fetchColumns(table);
+      setColumns(cols);
+    };
+    getColumns();
```
Comment (JackFratto) [inline src/renderer/components/execute/conditions/ConditionForm.tsx:118]:
```suggestion
    clearForm();
    clearErrors();
    getColumns();
```
So operation isn't maintained when switching table if options are already selected

File: src/renderer/components/execute/ParameterContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}: IParameterContainerProps) => {
+  if (activeParameters.length > 0) {
+    return (
+      <div className="condition-box">
+        <h4> Set Procedure Parameters </h4>
+        {activeParameters.map(
+          (item) =>
+            item.direction === 'IN' && (
+              <InputGroup className="mb-2" key={`${item.name}`}>
+                <InputGroup.Text bsPrefix="paramater-text">
```
Comment (JackFratto) [inline src/renderer/components/execute/ParameterContainer.tsx:21]:
```suggestion
                <InputGroup.Text>
```
forgot to take this out whoops

File: src/renderer/components/execute/conditions/ConditionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  }, [table]);
+
+  if (!form.level) {
+    return (
+      <Dropdown className="operation-dropdown">
+        <Dropdown.Toggle variant="primary">
+          {form.level || 'op type'}
```
Comment (JackFratto) [inline src/renderer/components/execute/conditions/ConditionForm.tsx:125]: this will always return 'op type' because of #L121 right?

Comment (alex-hobby) [reply src/renderer/components/execute/conditions/ConditionForm.tsx:125]: you're right. this is actually duplicate code with lines 143-157, i was just trying to get code pushed. The best way to do a lot of the conditional logic in this file is to split it all up into different components, which we should definitely do in the future time permitting.

File: src/renderer/types.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    | RowStringOperations
+    | RowBooleanOperations
+    | RowNumberOperations
+    | RowIDOperations;
+  result?: boolean;
+  expectedRecordMatches?: RecordMatches;
+  expectedNumRecords?: number;
+  total?: boolean;
+  table?: string;
+};
```
Comment (adityavidyadharan) [inline src/renderer/types.ts:26]: Why do we have this type? Seems to just be a reduced version of the UnitTestType

**Review by adityavidyadharan — CHANGES_REQUESTED** [review]
Overall changes look great. I think we'll need to add some stuff to help the user just because they probably won't understand what these conditions mean but that can be done in a future sprint/with feedback from KPMG.

Only real comment was about the type you created.

File: src/renderer/types.ts (hunk 4 above)
Comment (adityavidyadharan) [reply src/renderer/types.ts:26]: Think I see why you needed it but I don't think we need this type. 

We can use the built-in [`Partial`](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) type which just makes all fields optional. This way we don't have to manage two types and keep them in sync

Comment (alex-hobby) [reply src/renderer/types.ts:26]: It made my life easier.. not my proudest coding moment, and DEFINITELY is a top priority to figure out how to fix. I was running into a lot of annoyances with some of the values being required in the form. If you have a simple fix feel free to push.

**Review by insaan21 — APPROVED** [review]
lgtm

**Review by adityavidyadharan — APPROVED** [review]

**Pull request merged**